forwarder/src/lib.rs

```diff
@@ -127,3 +145,31 @@ pub trait ExtFeesCalculator {
         target_address: &str,
     ) -> U128;
 }
+
+// Validate that calculated part of the fee isn't more than `MAX_FEE_PERCENT`.
+fn is_fee_allowed(amount: U128, fee: U128) -> bool {
```

Comment:
This code is a bit difficult to read (IMHO), maybe plain

```
let percent = fee.0 * 100 / amount.0;
let remainder = (fee.0 * 100) % amount.0;

let allowed_percent = percent < MAX_FEE_PERCENT;
let no_remainder = percent == MAX_FEE_PERCENT && remainder == 0;

allowed_percent && no_remainer
```

Reply:
Ah, my bad, looking at this second time it's not gonna work this way!

Reply:
Yeah, probably match is OK, but this `zip` looks a bit unnecessary

Reply:
Just for the creation of a combined variable. Provide your variant please?

Reply:
Maybe just a plain wrap in `match()` constructor

```
    let fee = fee.0 * 100;
    let amount = amount.0;

    match (fee.checked_div(amount), fee.checked_rem(amount)) {
        (Some(percent), _) if percent > 100 => false,
        (Some(percent), Some(reminder)) if percent == 100 && reminder > 0 => false,
        _ => true,
    };
```

Reply:
Looks good to me.